src/uu/stat/src/stat.rs

```diff
@@ -299,6 +301,16 @@ fn group_num(s: &str) -> Cow<'_, str> {
     res.into()
 }
 
+/// Keeps major part of an integer
```

Comment:
The `major()` and `minor()` functions use incorrect bit-masking (`(n >> 8) & 0xFF` and `n & 0xFF`) that only handles old-style 8-bit device numbers. Modern Linux uses complex encoding with 12-bit major (up to 20-bit) and 8-bit minor (up to 32-bit) numbers. See `src/uu/mknod/src/mknod.rs:30-33` for the correct implementation

maybe move this function into uucore fs ?

Reply:
yeah